src/queries/labelText.ts

```diff
@@ -11,14 +14,38 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+const matchAccessibilityLabelledByText = (
+  rootInstance: ReactTestInstance,
+  nativeID: string | undefined,
+  text: TextMatch
+) => {
+  if (!nativeID) {
+    return false;
+  }
+
+  return (
+    rootInstance.findAll(
+      (node) =>
+        filterNodeByType(node, Text) &&
+        matchStringProp(node.props.nativeID, nativeID) &&
+        getNodeByText(node, text)
+    ).length > 0
+  );
+};
+
 const queryAllByLabelText = (
   instance: ReactTestInstance
 ): ((text: TextMatch) => Array<ReactTestInstance>) =>
   function queryAllByLabelTextFn(text) {
     return instance.findAll(
       (node) =>
         typeof node.type === 'string' &&
-        matchStringProp(node.props.accessibilityLabel, text)
+        (matchStringProp(node.props.accessibilityLabel, text) ||
+          matchAccessibilityLabelledByText(
+            instance,
```

Comment:
I don't like passing `instance` object here, but I couldn't figure out a better way, as I have to iterate over the whole tree to find an element with particular `nativeID`.

Reply:
I think it's fine as is. We have to find the matching element somehow, and that requires starting from the root.